main_spec.tex

```diff
@@ -103,10 +101,16 @@ \subsubsection{\textbf{SHPCLMOVE}}\label{subsec:shpclmove}
 \subsubsection{\textbf{SHPDEALLOC}}\label{subsec:shpdealloc}
 \input{content/shpdealloc.tex}
 
+\subsection{Thread Support}
+\input{content/threads_intro.tex} 
+\label{subsec:thread_support}
```

Comment:
Labels used wrong here. These labels should be after the section heading (`subsection`, `subsubsection`) and not after `\input`. Same applies for the two new API functions below. It doesn't affect the actual bookmarks in the PDF document, but better safe than sorry if whatever-is-in-input's-tex-document ever changes.